src/Engine/Combat/CombatEngine.java

```diff
@@ -1,18 +1,32 @@
 package Engine.Combat;
 
 import Terrain.GameMap;
+import Terrain.MapMaster;
 import Units.Unit;
 
 public class CombatEngine
 {
   /**
    * Static function to calculate the outcome of a battle between two units. It builds an
    * object to represent the specific combat instance and returns the result it calculates.
+   * Requires perfect map info just in case the COs need to get weird.
    */
-  public static BattleSummary calculateBattleResults( Unit attacker, Unit defender, GameMap map, int attackerX, int attackerY )
+  public static BattleSummary calculateBattleResults( Unit attacker, Unit defender, MapMaster map, int attackerX, int attackerY )
   {
     // Set up our combat scenario.
-    BattleInstance params = new BattleInstance(attacker, defender, map, attackerX, attackerY);
+    BattleInstance params = new BattleInstance(attacker, defender, map, attackerX, attackerY, false);
+
+    return params.calculateBattleResults();
+  }
+
+  /**
+   * Assuming Commanders get weird, this allows for you to check the results of combat without perfect map info.
+   * This also provides un-capped damage estimates, so perfect HP info isn't revealed by the map.
```

Comment:
Is there a reason we don't just do un-capped damage estimates by default? If I'm trying to attack an Infantry[4], and my readout says I can do 32% damage (because that's precisely how much HP it has left), I can't tell if that will do the job. I'd rather it just say I could do 45% damage so I know where I stand. I'm not sure I want a separate "simulate battle" flag to complicate things, especially since the fact you are building an attack action against something means you can see it, and therefore know its strength (the exception here is Commanders who can hide unit HP, but those would just always be shown as full-strength).

Reply:
I guess I can see a justification for a `simulate` version to ignore luck (not that we support that), where a call to `calculate` will return different results each time. Hmm, perhaps.